ui/src/data/DataStoreBase.ts

```diff
@@ -0,0 +1,168 @@
+import ZomeApi from "../api/zomeApi";
+import { getZomeApi } from "../hcWebsockets";
+import { RustNode, ThingInput } from "../types/holochain";
+import { getLastPart, getParentPath, getRustNodePath, PathedData } from "./models/PathedData";
+import { ObjectTransformations } from "./models/ObjectTransformations";
+import { Root, RootShape } from "./models/Application/Root";
+
+/**
+ * Data store object
+ */
+
+export class DataStoreBase {
```

Comment:
I'm not sure I understand the distinction between DataStore and DataStoreBase.

Reply:
It's a very mild distinction. DataStoreBase is the generic, could be used with any PathedData object and could be rolled out into a separate library for others to use. DataStore could also have been called AppDataStore and it has functions which are specific for our application, like getting lists of objects to display in dropdowns or the sidebars.